src/dslite_parser.rs

```diff
@@ -213,19 +223,24 @@ fn parse_field(el: &Element) -> Field {
     assert_eq!(el.name, "bitfield");
 
     let name = uw!(el.attributes.get("id")).to_owned();
-    let description = uw!(el.attributes.get("description")).to_owned();
+    let mut description = uw!(el.attributes.get("description")).to_owned();
     let begin = uw!(utils::parse_u32(uw!(el.attributes.get("begin"))));
     let end = uw!(utils::parse_u32(uw!(el.attributes.get("end"))));
     let width = uw!(utils::parse_u32(uw!(el.attributes.get("width"))));
 
+    if description.trim().is_empty() {
+        description = name.clone();
+    }
+
     assert_eq!((begin as isize - end as isize).abs() + 1, width as isize);
 
     let offset = ::std::cmp::min(begin, end);
 
     let rwa = uw!(el.attributes.get("rwaccess")).to_owned();
-    assert!(rwa == "R/W" || rwa == "RW");
+    assert!(rwa == "R/W" || rwa == "RW" || rwa == "R");
```

Comment:
If we now have readonly fields, maybe we need to propagate this info down the line, so it is also marked readonly in svd file?

Reply:
will do

Reply:
thanks